# 哨兵上云台
MV_Sentry:
    #内参矩阵
    Intrinsic: [3626.5098590324505,  0.000000000000,  689.1283950373454,
                0.000000000000,  3626.285156056285,  582.7004766963166,
                0.000000000000,  0.000000000000,  1.000000000000]
    #畸变参数矩阵，k1,k2,k3,p1,p2。一定注意是k1,k2,k3,p1,p2!!!和ROS标出来的k1,k2,p1,p2,k3不同！！！
    Coeff: [-0.04251601401350153, 0.963321415304225, 0.0, -0.0009602427973179668, -0.006641239457517189]
    #x,y,z坐标偏移量(相机至枪管),单位m
    xyz_offset: [0,-0.05,0]
    #yaw,pitch角度偏移量,单位度
    angle_offset: [-1.5, -1.75]
    #使用迭代法求解pitch补偿的最大迭代次数
    max_iter: 10
    #停止迭代的最小误差(单位m)
    stop_error: 0.001
    #龙格库塔法求解落点的迭代次数
    R_K_iter: 60

    #陀螺仪到相机的变换矩阵 R_c = T_ci * R_i
    T_ci: [ 0.02617095,   0.99945437,  -0.02015029,  -0.00043464,
            0.1181205,   -0.02310774,  -0.99273037,   0.00326701,
            -0.99265433,  0.02360054, -0.1186608,  -0.00323944,
            0,          0,          0,          1        ]
    #相机到陀螺仪的变换矩阵 R_i = T_ic * R_c
    T_ic: [ 0.02617095,   0.1181205,   -0.99265433, -0.00359017,
            0.99945437,  -0.02310774,   0.02360054,  0.00058635,
            -0.02015029, -0.99273037,  -0.1186608,   0.00285011,
            0,          0,          0,         1       ]
    #陀螺仪至云台转轴中心平移向量 R_g = R_i - T_iw
    T_iw: [-0.03, -0.05, -0.02]

# 4号步兵
KE0200110075:
    #陀螺仪坐标系：枪口反方向为x轴正向，弹舱舵机右侧为y轴正向，垂直向上为z轴正向
    #内参矩阵
    Intrinsic:  [1675.042871,  0.000000000000,  623.481121,
                0.000000000000,1674.696037, 556.002821,
                0.000000000000,  0.000000000000,  1.000000000000]
    # Coeff: [-0.058880415613303524, 0.09971916846980647, 0, -0.002214233751008686, 0.0003401562523311091]
    Coeff: [-0.06719080320827164, 0.08771589756714641, 0.0, 0.0009056377800116876, -0.004604447207374161] 
    #x,y,z坐标偏移量(相机至枪管),单位m
    xyz_offset: [0,0,0]
    #yaw,pitch角度偏移量,单位度 left- down-
    angle_offset: [0,0]
    #使用迭代法求解pitch补偿的最大迭代次数
    max_iter: 10
    #停止迭代的最小误差(单位m)
    stop_error: 0.001
    #龙格库塔法求解落点的迭代次数
    R_K_iter: 50
    #陀螺仪到相机的变换矩阵 R_c = T_ci * R_i
    # T_ci:  [-0.04923536,0.99832596,0.03035069,-0.0001569, 
    #         -0.00286999,0.03024601,-0.99953836,0.00126626,
    #         -0.99878308,-0.04929974,0.00137601,-0.2468961,
    #          0.        ,0.        ,0.         ,1.          ]
    T_ci:  [-0.00687597 , 0.9999657 ,  0.0046184  , 0.00935481,
            0.0005452  , 0.00462226, -0.99998917 , 0.00111573,
            -0.99997621 ,-0.00687337, -0.00057697 ,-0.00203505,
            0         , 0        ,  0         , 1.        ]
    #相机到陀螺仪的变换矩阵 R_i = T_ic * R_c
    # T_ic: [-0.04923536,-0.00286999,-0.99878308,-0.2466366,
    #         0.99832596, 0.03024601,-0.04929974,-0.00109885,
    #         0.03035069,-0.99953836, 0.00137601, 0.00130441,
    #         0.        , 0.        , 0.        , 1.        ]
    T_ic: [-0.00687597,  0.0005452 , -0.99997621, -0.00197128,
            0.9999657 ,  0.00462226, -0.00687337, -0.00937364,
            0.0046184 , -0.99998917, -0.00057697, 0.00107134,
            0         ,  0         , 0          ,  1.        ]
    #陀螺仪至云台转轴中心平移向量 R_g = R_i - T_iw
    T_iw: [-0.1, 0, -0.05]

# 英雄
KE0200110073:
    #内参矩阵
    Intrinsic: [1746.05537, 0.000000000000,  602.87169,
                0.000000000000, 1746.45998,   491.51679,
                0.000000000000,  0.000000000000,  1.000000000000]
    #畸变参数矩阵，k1,k2,k3,p1,p2。一定注意是k1,k2,k3,p1,p2!!!和ROS标出来的k1,k2,p1,p2,k3不同！！！
    Coeff: [-0.059401, 0.357116, 0.0, -0.001874, -0.007872]
    #x,y,z坐标偏移量(相机至枪管),单位m
    xyz_offset: [0,0,0]
    #yaw,pitch角度偏移量,单位度
    angle_offset: [-1.65, -2.65]
    #使用迭代法求解pitch补偿的最大迭代次数
    max_iter: 10
    #停止迭代的最小误差(单位m)
    stop_error: 0.001
    #龙格库塔法求解落点的迭代次数
    R_K_iter: 50
    #陀螺仪到相机的变换矩阵 R_c = T_ci * R_i
    T_ci: [ -0.04201099,  0.99766553, -0.05383838, -0.06218852,
            0.04260072, -0.05204827, -0.99773551,  0.04941592,
           -0.99820852, -0.04420941, -0.04031467, -0.01211039,  
            0          ,0           ,0            ,1        ]
    #相机到陀螺仪的变换矩阵 R_i = T_ic * R_c
    T_ic: [ -0.04201099,  0.04260072, -0.99820852, -0.01680645,
             0.99766553, -0.05204827, -0.04420941,  0.06407996,
           -0.05383838, -0.99773551, -0.04031467,  0.04546766,
            0          ,0           ,0           ,1        ]
    #陀螺仪至云台转轴中心平移向量 R_w = R_i - T_iw
    T_iw: [0.02, 0.01, -0.05]
   
# 3号步兵
KE0200110074:
    #内参矩阵
    Intrinsic: [1673.424472385629,  0.000000000000,  637.8832545589809,
                0.000000000000,  1675.7985620485485,  504.2286845853391,
                0.000000000000,  0.000000000000,  1.000000000000]
    #畸变参数矩阵，k1,k2,k3,p1,p2。一定注意是k1,k2,k3,p1,p2!!!和ROS标出来的k1,k2,p1,p2,k3不同！！！
    Coeff: [-0.058880415613303524, 0.09971916846980647, 0, -0.002214233751008686, 0.0003401562523311091]
    #x,y,z坐标偏移量(相机至枪管),单位m
    xyz_offset: [0,0,0]
    #yaw,pitch角度偏移量,单位度
    angle_offset: [-0.79, 7.7]
    #使用迭代法求解pitch补偿的最大迭代次数
    max_iter: 10
    #停止迭代的最小误差(单位m)
    stop_error: 0.001
    #龙格库塔法求解落点的迭代次数
    R_K_iter: 50
    #陀螺仪到相机的变换矩阵 R_c = T_ci * R_i
    T_ci: [ 0.00741479 ,0.99653649  ,-0.08282545 ,0.01111486,
           -0.04520383 ,-0.08240902 ,-0.99557288 ,0.00670747,
           -0.99895027 ,0.01112599  ,0.04443622  ,-0.077358,  
            0          ,0           ,0            ,1        ]
    #相机到陀螺仪的变换矩阵 R_i = T_ic * R_c
    T_ic: [ 0.00741479 ,-0.04520383 ,-0.99895027 ,-0.07705601,
            0.99653649 ,-0.08240902 , 0.01112599 ,-0.00966292,
           -0.08282545 ,-0.99557288 , 0.04443622 , 0.01103587,
            0          ,0           ,0           ,1        ]
    #陀螺仪至云台转轴中心平移向量 R_w = R_i - T_iw
    T_iw: [0, 0, -0.02]

# 5号blancer步兵
KE0200110076:
    #内参矩阵
    Intrinsic: [1656.55423,  0.000000000000,  691.49622,
                0.000000000000, 1659.44139, 482.0145,
                0.000000000000,  0.000000000000,  1.000000000000]
    #畸变参数矩阵，k1,k2,k3,p1,p2。一定注意是k1,k2,k3,p1,p2!!!和ROS标出来的k1,k2,p1,p2,k3不同！！！
    Coeff: [-0.083215, 0.179504, 0.0, -0.001395, 0.004738]
    #x,y,z坐标偏移量(相机至枪管),单位m
    xyz_offset: [0,0,0]
    #yaw,pitch角度偏移量,单位度
    angle_offset: [2.2, 2.6]
    #使用迭代法求解pitch补偿的最大迭代次数
    max_iter: 10
    #停止迭代的最小误差(单位m)
    stop_error: 0.001
    #龙格库塔法求解落点的迭代次数
    R_K_iter: 50
    #陀螺仪到相机的变换矩阵 R_c = T_ci * R_i
    T_ci: [-0.015054,  -0.99745114, -0.06974674, 0.00272642,
           -0.02372202, 0.07009129, -0.99725848, 0.00051117,
           0.99960524, -0.0133582,  -0.02471671, 0.00006743,
           0.0,        0.0,         0.0,         1.0  ]
    #相机到陀螺仪的变换矩阵 R_i = T_ic * R_c
    T_ic: [-0.015054,   -0.02372202, 0.99960524,  -0.00001424,
           -0.99745114, 0.07009129,  -0.0133582,  0.00268454,
           -0.06974674, -0.99725848, -0.02471671, 0.0007016,
           0.0,         0.0        ,  0.0       , 1.0        ]
    #陀螺仪至云台转轴中心平移向量 R_w = R_i - T_iw
    T_iw: [0, 0, -0.02]